### PR TITLE
More rename of service to application, mainly in state

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

A mechanical change to clean up most of the remaining places where we use "service" instead of "application". The bulk are in state.
There's also a few file renames where the file name included "service".

There was one latent bug fixed as part of this - the allwatcher api was ordering deltas using "service" but the backend was producing records with "application".

## QA steps

Smoke test using a bootstrap and deploy.

### CODE REVIEW

File: api/base/types.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ type ModelStatus struct {
 	TotalMachineCount  int
 	CoreCount          int
 	HostedMachineCount int
-	ServiceCount       int
+	ApplicationCount   int
```
Comment (anastasiamac) [inline api/base/types.go]: Considering that this changes the name in the returning structure, I think that the implications are further reaching - facades need to be bumped, transformation code at presentation layer changed, etc...

File: apiserver/resources.go [hunk 2]
```diff
@@ -164,7 +164,7 @@ func (h *ResourcesHandler) readResource(backend ResourcesBackend, req *http.Requ
 		return nil, errors.Trace(err)
 	}
 	return &uploadedResource{
-		Service:   uReq.Service,
+		Service:   uReq.Application,
```
Comment (anastasiamac) [inline apiserver/resources.go]: If you go to this length, maybe worth renaming 'Service' here too?

File: resource/api/upload.go [hunk 3]
```diff
@@ -18,8 +18,8 @@ import (
 
 // UploadRequest defines a single upload request.
 type UploadRequest struct {
-	// Service is the application ID.
-	Service string
+	// Application is the application ID.
+	Application string
```
Comment (anastasiamac) [inline resource/api/upload.go]: I think this renaming of property also means you'd need to bump facades that use/return this struct.

In fact, I wonder if we need to "deprecate" it - still carry around "Service" but keep it empty and use "Application" everywhere that matters.

File: state/presence/presence_test.go [hunk 4]
```diff
@@ -483,7 +483,7 @@ func (s *PresenceSuite) TestSync(c *gc.C) {
 	}
 }
 
-func (s *PresenceSuite) TestTwoEnvironments(c *gc.C) {
+func (s *PresenceSuite) TestTwomodels(c *gc.C) {
```
Comment (anastasiamac) [inline state/presence/presence_test.go]: Capital M for models, please :D

File: resource/api/upload.go (hunk 3 above)
Comment (wallyworld) [reply resource/api/upload.go]: This is also not a struct send over the wire. It's used internally in the api and apiserver layers to encapsulate data - the content comes from unmarshalling a http request.

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**